src/project-css/app/Adapters/ImportAdapter.php

```diff
@@ -0,0 +1,208 @@
+     * the previous line was saved and merge them and check the count again.
+     * if their is insufficent items we will save the tkns and the row position
+     * so we can attempt a merge later.
+     * @param string $curLine current line read from the file
+     * @param string $delimiter type of delmiter that is used in the file
+     * @param integer $orgCount current number of fields expected
+     * @param integer $prcssd current position in the file
+     * @return string
+     */
+    public function prepareLine($curLine, $delimiter, $orgCount, $prcssd) {
+        // Strip out Quotes that are sometimes seen in csv files around each item
+        $curLine = str_replace('"', "", $curLine);
+
+        // Tokenize the line
+        $this->tkns = (new CSVHelper)->tknze($curLine, $delimiter);
+
+        // Validate the tokens and filter them
+        $this->tkns = (new CSVHelper)->fltrTkns($this->tkns);
+
+        // if lastErrRow is the previous row try to combine the lines
+        if ((count($this->tkns) != $orgCount) && ($this->lastErrRow == $prcssd) && ($this->savedTkns != NULL)) {
+            $this->mergeLines();
+
+            // clear last saved line since we did a merge
+            $this->lastErrRow = NULL;
+            $this->savedTkns = NULL;
+        }
+
+        // if token count doesn't match what is exptected save the tkns and last row position
+        if (is_array($this->tkns) && (count($this->tkns) != $orgCount)) {
+          // save the last row position and the Tokenized row
+          $this->lastErrRow = $prcssd;
+          $this->savedTkns = $this->tkns;
+          return false;
+        }
+
+        return true;
+    }
+
+    /**
+    * takes array of tokens. Creates a search index and 
+    * inserts them into the table
+    * 
+    * @param array $tkns array containing all fields of record
+    * @param integer $orgCount number of expected fields in the record
+    * @param object $clmnLst array containing all field names for table
+    * @return boolean
+    */   
+    public function processLine($orgCount) {
+        if(!is_array($this->tkns) && empty($this->tkns)) { return false; }
+
+        // verify that passed $tkns match the expected field count
+        if (count($this->tkns) == $orgCount) {
+          // Declae an array
+          $this->curArry = array();
+
+          // Compact them into one array with utf8 encoding
+          for ($i = 0; $i<$orgCount; $i++) {
+            $this->curArry[ strval($this->clmnLst[ $i ]) ] = utf8_encode($this->tkns[ $i ]);
+          }
+
+        }
+    }
+
+    /**
+     * Process employs following algorithm:
+     * get all the column names from table name
+     * 1. Read the file as spl object
+     * 2. For each line
+     *   1. Validate
+     * @param string $tblNme
+     * @param string $fltFlePath
+     * @param string $fltFleNmetype
+     * @param boolean $ignoreFirst 
+     * @return void
+     */
+    public function process($tblNme, $fltFlePath, $fltFleNme) {
```

Comment:
THIS IS A MASSIVE FUNCTION!

Reply:
Thinking about better OOP on this the documentation says it all.  You are trying to do the following in a single function 

1. Get all column names from the table name
2. Read the file as an SPL
3. Validate Each Line
4. Set a large amount of instance variables
5. Save Data Or Throw an Exception

This completely ignores single responsibility methods.